spyder/widgets/editor.py

```diff
@@ -0,0 +1,549 @@
+    def __init__(self, code_editor=None):
+        super().__init__(code_editor)
+
+        self._editor = code_editor
+        self._left_padding = 6  # Pixels
+        self._right_padding = 3  # Pixels
+
+    # --- Qt overrides
+    # ------------------------------------------------------------------------
+    def sizeHint(self):
+        return QSize(self._editor.linenumberarea_width(), 0)
+
+    def paintEvent(self, event):
+        self._editor.linenumberarea_paint_event(event)
+
+
+class SimpleCodeEditor(QPlainTextEdit, BaseEditMixin):
+    """Simple editor with highlight features."""
+
+    LANGUAGE_HIGHLIGHTERS = {
+        'Python': (sh.PythonSH, '#'),
+        'Cython': (sh.CythonSH, '#'),
+        'Fortran77': (sh.Fortran77SH, 'c'),
+        'Fortran': (sh.FortranSH, '!'),
+        'Idl': (sh.IdlSH, ';'),
+        'Diff': (sh.DiffSH, ''),
+        'GetText': (sh.GetTextSH, '#'),
+        'Nsis': (sh.NsisSH, '#'),
+        'Html': (sh.HtmlSH, ''),
+        'Yaml': (sh.YamlSH, '#'),
+        'Cpp': (sh.CppSH, '//'),
+        'OpenCL': (sh.OpenCLSH, '//'),
+        'Enaml': (sh.EnamlSH, '#'),
+        'Markdown': (sh.MarkdownSH, '#'),
+        # Every other language
+        'None': (sh.TextSH, ''),
+    }
+
+    # --- Signals
+    # ------------------------------------------------------------------------
+    sig_focus_changed = Signal()
+    """
+    """
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        # Variables
+        self._linenumber_enabled = None
+        self._color_scheme = "spyder/dark"
+        self._language = None
+        self._blanks_enabled = None
+        self._scrollpastend_enabled = None
+        self._wrap_mode = None
+        self.supported_language = False
+
+        # Widgets
+        self._highlighter = None
+        self.linenumberarea = LineNumberArea(self)
+
+        # Widget setup
+        self.setObjectName(self.__class__.__name__ + str(id(self)))
+        self.update_linenumberarea_width(0)
+        self._highlight_current_line()
+
+        # Signals
+        self.blockCountChanged.connect(self.update_linenumberarea_width)
+        self.updateRequest.connect(self.update_linenumberarea)
+        self.cursorPositionChanged.connect(self._highlight_current_line)
+
+    # --- Private API
+    # ------------------------------------------------------------------------
+    def _apply_color_scheme(self):
+        hl = self._highlighter
+        hl.set_color_scheme(self._color_scheme)
+        self._set_palette(background=hl.get_background_color(),
+                          foreground=hl.get_foreground_color())
+
+    def _set_palette(self, background, foreground):
+        style = ("QPlainTextEdit#%s {background: %s; color: %s;}" %
+                 (self.objectName(), background.name(), foreground.name()))
+        self.setStyleSheet(style)
+        self.rehighlight()
+
+    def _highlight_current_line(self):
+        if self._highlighter:
+            extraSelections = []

```

Comment:
Please don't use camel case completions. We don't use them anywhere else in our code.

Reply:
Fixed.